src/TypeScript/DefinitionWriter.cpp

```diff
@@ -654,7 +698,7 @@ std::string DefinitionWriter::tsifyType(const Type& type)
             return "string";
         }
         else if (interface.name == "NSDate") {
```

Comment:
As I think of it factory functions return (instacetype) instead of NSDate so this should not be a concern. Can we keep NSDate here so we can integrate the new definitions in the modules now and log this for future improvement?